File: PlaceYourBets/Container.vb
Imports System
Imports System.IO
Imports System.Net
Imports System.Text

Public Class ContainerForm

    Private fixtureNo As Integer = 0
    Private fixtures As New List(Of Fixture)
    Private fixtureBoxList As New List(Of FixtureBox)
    Private userList As New List(Of String)
    Private submittedList As New List(Of String)
    Private userResultList As New List(Of Users)
    Private fix As Integer

    Public Property currentUser As Integer
        Get
            Return m_user
        End Get
        Set(value As Integer)
            m_user = value
        End Set
    End Property
    Private m_user As Integer

    Private Sub ContainerForm_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

        Dim version As Version
        version = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version
        Me.Text = String.Format("v{0}.{1}", version.Major, version.Minor)

        Me.enterPredictionsButton.Visible = True
        Me.betSlipButton.Visible = True
        Me.whoBetButton.Visible = True
        Me.predictionLeagueButton.Visible = True
        Me.myAccountButton.Visible = True
        Me.logOutButton.Visible = True
        Me.enterPredictionsButton.Text = ""
        Me.betSlipButton.Text = ""
        Me.whoBetButton.Text = ""
        Me.predictionLeagueButton.Text = ""
        Me.myAccountButton.Text = ""
        Me.logOutButton.Text = ""
        Me.enterPredictionsButton.Enabled = False
        Me.betSlipButton.Enabled = False
        Me.whoBetButton.Enabled = False
        Me.predictionLeagueButton.Enabled = False
        Me.myAccountButton.Enabled = False
        Me.logOutButton.Enabled = False
        Me.submitGroupBox.Visible = True
        Me.submitGroupBox.Visible = False
        Me.submitButton.Visible = False
        Me.editButton.Visible = False
        Me.MainFlowLayoutPanel.Visible = False
        Me.MainFlowLayoutPanel.Enabled = False
        Me.MainFlowLayoutPanel.Controls.Clear()

        Dim fLoginUserControl As New LoginUserControl
        Me.MainFlowLayoutPanel.Controls.Add(fLoginUserControl)
        Me.MainFlowLayoutPanel.Visible = True
        Me.MainFlowLayoutPanel.Enabled = True
        Me.AcceptButton = fLoginUserControl.loginButton

    End Sub


    Private Sub enterPredictionsButton_Click(sender As System.Object, e As System.EventArgs) Handles enterPredictionsButton.Click

        fixtureBoxList.Clear()
        Me.MainFlowLayoutPanel.Controls.Clear()
        Dim fb As New FixtureBuilder
        fixtures = fb.getFixtures()
        fix = 0

        Try

        For Each Fixture As Fixture In fixtures
            Dim usercontrol As New FixtureBox(Fixture.Home_Team, Fixture.Away_Team, Fixture.Date, getTeamBadge(fixtures.Item(fix).Home_Team), getTeamBadge(fixtures.Item(fix).Away_Team))
            MainFlowLayoutPanel.Controls.Add(usercontrol)
            usercontrol.Dock = DockStyle.Fill
            usercontrol.Visible = False
            fixtureBoxList.Add(usercontrol)
            fix = fix + 1

        Next

        fixtureBoxList.Item(0).Visible = True
        fixtureBoxList.Item(0).backButton.Enabled = False
        Me.submitButton.Visible = False
            Me.editButton.Visible = False

        Catch ex As NullReferenceException
            MsgBox("No fixtures available")
        End Try

        If fix <> 10 Then
            Me.MainFlowLayoutPanel.Controls.Clear()
            fixtureBoxList.Clear()
            MsgBox("There are currently no fixtures to display")
        End If

    End Sub

    Private Sub submitButton_Click(sender As System.Object, e As System.EventArgs) Handles submitButton.Click

        ProgressBarForm.Show()

        Me.SubmitPredictions()

        ProgressBarForm.Close()

        MsgBox("Predictions submitted")
        submitGroupBox.Visible = False
        Me.MainFlowLayoutPanel.Controls.Clear()


    End Sub

    Private Sub editButton_Click_1(sender As System.Object, e As System.EventArgs) Handles editButton.Click
        Dim list As New List(Of Control)

        For Each control In MainFlowLayoutPanel.Controls
            If TypeOf control Is FixtureReviewUserControl Then
                list.Add(control)
            End If
        Next

        For Each control In list
            MainFlowLayoutPanel.Controls.Remove(control)

        Next

        For Each fixture As Fixture In fixtures
            Dim review As New FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team)
            Me.MainFlowLayoutPanel.Controls.Remove(review)

        Next

        fixtureBoxList.Item(0).Visible = True
        fixtureBoxList.Item(0).backButton.Enabled = False
        fixtureNo = 0

        Me.submitGroupBox.Visible = False
        Me.submitButton.Visible = False
        Me.editButton.Visible = False
    End Sub

    Private Sub betSlipButton_Click_1(sender As System.Object, e As System.EventArgs) Handles betSlipButton.Click
        Me.MainFlowLayoutPanel.Controls.Clear()
        Dim fb As New FixtureBuilder
        fixtures = fb.getBet()

        Me.submitGroupBox.Visible = False

        Try
            For Each fixture As Fixture In fixtures
                Dim review As New FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team)
                Me.MainFlowLayoutPanel.Controls.Add(review)

                If fixture.Prediction = 1 Then
                    review.homeFixtureLabel.BackColor = Color.LightGreen
                End If
                If fixture.Prediction = 2 Then
                    review.awayFixtureLabel.BackColor = Color.LightGreen
                End If
                If fixture.Prediction = 3 Then
                    review.drawLabel.BackColor = Color.LightGreen
                End If
            Next

        Catch ex As NullReferenceException
            MsgBox("Not enough users have entered their predictions yet. Please try later")
        End Try
    End Sub

    Private Sub whoBetButton_Click(sender As System.Object, e As System.EventArgs) Handles whoBetButton.Click
        Me.MainFlowLayoutPanel.Controls.Clear()
        Dim fb As New FixtureBuilder
        userList = fb.getUsers()
        submittedList = fb.usersSubmitted()

        Me.MainFlowLayoutPanel.Enabled = True
        Me.MainFlowLayoutPanel.Visible = True
        Me.submitGroupBox.Visible = False

        For Each user As String In userList
            Dim review As New UserPredictionsUserControl(user)
            Me.MainFlowLayoutPanel.Controls.Add(review)

            If submittedList.Contains(user) Then
                review.submittedLabel.BackColor = Color.LightGreen
                review.submittedLabel.Text = "Submitted"
            End If

        Next
    End Sub

    Private Sub predictionLeagueButton_Click(sender As System.Object, e As System.EventArgs) Handles predictionLeagueButton.Click
        Me.MainFlowLayoutPanel.Controls.Clear()
        Dim fb As New FixtureBuilder
        userResultList = fb.getScores()
        Dim userDictionary As New Dictionary(Of Integer, String)
        userDictionary = fb.getIdsandUsers()

        Me.MainFlowLayoutPanel.Enabled = True
        Me.MainFlowLayoutPanel.Visible = True
        Me.submitGroupBox.Visible = False

        Dim name As String = ""

        For Each user As Users In userResultList
            userDictionary.TryGetValue(user.UserID, name)
            Dim review As New PredictionLeagueUserControl(name, user.correct)
            Me.MainFlowLayoutPanel.Controls.Add(review)

        Next
    End Sub

    Private Sub logOutButton_Click(sender As System.Object, e As System.EventArgs) Handles logOutButton.Click
        Me.MainFlowLayoutPanel.Controls.Clear()
        Me.enterPredictionsButton.Text = ""
        Me.betSlipButton.Text = ""
        Me.whoBetButton.Text = ""
        Me.predictionLeagueButton.Text = ""
        Me.myAccountButton.Text = ""
        Me.logOutButton.Text = ""
        Me.enterPredictionsButton.Enabled = False
        Me.betSlipButton.Enabled = False
        Me.whoBetButton.Enabled = False
        Me.predictionLeagueButton.Enabled = False
        Me.myAccountButton.Enabled = False
        Me.logOutButton.Enabled = False
        Me.coverTitleLabel.Visible = False
        Dim fLoginUserControl As New LoginUserControl
        Me.MainFlowLayoutPanel.Controls.Add(fLoginUserControl)
        Me.AcceptButton = fLoginUserControl.loginButton
    End Sub

    Public Sub makeFixtureVisible(forward As Boolean)
        Try
            If (forward) Then

                Dim previousFixtureNo As Integer = fixtureNo
                fixtureNo = fixtureNo + 1
                fixtureBoxList.Item(previousFixtureNo).Visible = False

                If fixtureNo = 9 Then
                    fixtureBoxList.Item(fixtureNo).nextButton.Text = "Review >"
                    fixtureBoxList.Item(fixtureNo).nextButton.BackColor = Color.LightGreen
                End If

                If fixtureBoxList.Item(previousFixtureNo).homeRadioButton.Checked = True Then
                    fixtures.Item(previousFixtureNo).Prediction = 1
                End If
                If fixtureBoxList.Item(previousFixtureNo).awayRadioButton.Checked = True Then
                    fixtures.Item(previousFixtureNo).Prediction = 2
                End If
                If fixtureBoxList.Item(previousFixtureNo).drawRadioButton.Checked = True Then
                    fixtures.Item(previousFixtureNo).Prediction = 3
                End If

            Else
                fixtureNo = fixtureNo - 1
                fixtureBoxList.Item(fixtureNo + 1).Visible = False

            End If

            fixtureBoxList.Item(fixtureNo).Visible = True

        Catch ex As ArgumentOutOfRangeException
            MsgBox("The full set of fixtures is not available. Please email complaints to michael.cole@advancedcomputersoftware.com")
        End Try
    End Sub

    Public Function getFixtureNo() As Integer
        Return fixtureNo
    End Function

    Public Sub reviewPredictions()

        Dim previousFixtureNo As Integer = fixtureNo
        fixtureNo = fixtureNo + 1
        fixtureBoxList.Item(fixtureNo - 1).Visible = False

        If fixtureBoxList.Item(previousFixtureNo).homeRadioButton.Checked = True Then
            fixtures.Item(previousFixtureNo).Prediction = 1
        End If
        If fixtureBoxList.Item(previousFixtureNo).awayRadioButton.Checked = True Then
            fixtures.Item(previousFixtureNo).Prediction = 2
        End If
        If fixtureBoxList.Item(previousFixtureNo).drawRadioButton.Checked = True Then
            fixtures.Item(previousFixtureNo).Prediction = 3
        End If

        Me.MainFlowLayoutPanel.Enabled = True
        Me.MainFlowLayoutPanel.Visible = True
        Me.submitGroupBox.Visible = True
        Me.submitButton.Visible = True
        Me.editButton.Visible = True


        For Each fixture As Fixture In fixtures
            Dim review As New FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team)
            Me.MainFlowLayoutPanel.Controls.Add(review)

            If fixture.Prediction = 1 Then
                review.homeFixtureLabel.BackColor = Color.LightGreen
            End If
            If fixture.Prediction = 2 Then
                review.awayFixtureLabel.BackColor = Color.LightGreen
            End If
            If fixture.Prediction = 3 Then
                review.drawLabel.BackColor = Color.LightGreen
            End If
        Next

    End Sub

    Public Function getTeamBadge(team As String) As Image
        Dim badge As Image = My.Resources.icon2
        Select Case team
            Case "Arsenal"
                badge = My.Resources.ARS
            Case "Aston Villa"
                badge = My.Resources.AST
            Case "Burnley"
                badge = My.Resources.BUR
            Case "Chelsea"
                badge = My.Resources.CHE
            Case "Crystal Palace"
                badge = My.Resources.CRY
            Case "Everton"
                badge = My.Resources.EVE
            Case "Hull City"
                badge = My.Resources.HUL
            Case "Leicester City"
                badge = My.Resources.LEI
            Case "Liverpool"
                badge = My.Resources.LIV
            Case "Manchester City"
                badge = My.Resources.MNC
            Case "Manchester United"
                badge = My.Resources.MNU
            Case "Newcastle United"
                badge = My.Resources.NWC
            Case "Queens Park Rangers"
                badge = My.Resources.QPR
            Case "Southampton"
                badge = My.Resources.SOU
            Case "Stoke City"
                badge = My.Resources.STO
            Case "Sunderland"
                badge = My.Resources.SUN
            Case "Swansea City"
                badge = My.Resources.SWA
            Case "Tottenham Hotspur"
                badge = My.Resources.TOT
            Case "West Bromwich Albion"
                badge = My.Resources.WBA
            Case "West Ham United"
                badge = My.Resources.WHU
            Case Else
                badge = My.Resources.icon2

        End Select

        Return badge

    End Function

    Public Sub SubmitPredictions()

        Dim submitted As Boolean = False

        While (Not submitted)

            For Each fixture As Fixture In fixtures

                ProgressBarForm.ProgressBar1.PerformStep()
                Dim UrlBase As String = "http://www.predictresults.co.uk/pradeep/InsertDataUsingSP.php?userid="
                Dim user As String = currentUser.ToString()
                Dim fixtureID As String = fixture.ID.ToString()
                Dim prediction As String = fixture.Prediction.ToString()
                Dim url As String = UrlBase + user + "&fixtureid=" + fixtureID + "&prediction=" + prediction
                Dim request As HttpWebRequest = CType(WebRequest.Create(url), HttpWebRequest)
                Dim response As HttpWebResponse = CType(request.GetResponse(), HttpWebResponse)

                'Get the stream associated with the response. 
                Dim receiveStream As Stream = response.GetResponseStream()
                Dim readStream As New StreamReader(receiveStream, Encoding.UTF8)

                response.Close()
                readStream.Close()

            Next

            submitted = True

        End While
    End Sub

    
End Class

File: PlaceYourBets.ConvertedToC#/Container.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.IO;
using System.Net;
using System.Text;
namespace PlaceYourBets
{

	public partial class ContainerForm
	{

		private int fixtureNo = 0;
		private List<Fixture> fixtures = new List<Fixture>();
		private List<FixtureBox> fixtureBoxList = new List<FixtureBox>();
		private List<string> userList = new List<string>();
		private List<string> submittedList = new List<string>();
		private List<Users> userResultList = new List<Users>();

		private int fix;
		public int currentUser {
			get { return m_user; }
			set { m_user = value; }
		}

		private int m_user;

		private void ContainerForm_Load(System.Object sender, System.EventArgs e)
		{
			Version version = null;
			version = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version;
			this.Text = string.Format("v{0}.{1}", version.Major, version.Minor);

			this.enterPredictionsButton.Visible = true;
			this.betSlipButton.Visible = true;
			this.whoBetButton.Visible = true;
			this.predictionLeagueButton.Visible = true;
			this.myAccountButton.Visible = true;
			this.logOutButton.Visible = true;
			this.enterPredictionsButton.Text = "";
			this.betSlipButton.Text = "";
			this.whoBetButton.Text = "";
			this.predictionLeagueButton.Text = "";
			this.myAccountButton.Text = "";
			this.logOutButton.Text = "";
			this.enterPredictionsButton.Enabled = false;
			this.betSlipButton.Enabled = false;
			this.whoBetButton.Enabled = false;
			this.predictionLeagueButton.Enabled = false;
			this.myAccountButton.Enabled = false;
			this.logOutButton.Enabled = false;
			this.submitGroupBox.Visible = true;
			this.submitGroupBox.Visible = false;
			this.submitButton.Visible = false;
			this.editButton.Visible = false;
			this.MainFlowLayoutPanel.Visible = false;
			this.MainFlowLayoutPanel.Enabled = false;
			this.MainFlowLayoutPanel.Controls.Clear();

			LoginUserControl fLoginUserControl = new LoginUserControl();
			this.MainFlowLayoutPanel.Controls.Add(fLoginUserControl);
			this.MainFlowLayoutPanel.Visible = true;
			this.MainFlowLayoutPanel.Enabled = true;
			this.AcceptButton = fLoginUserControl.loginButton;

		}



		private void enterPredictionsButton_Click(System.Object sender, System.EventArgs e)
		{
			fixtureBoxList.Clear();
			this.MainFlowLayoutPanel.Controls.Clear();
			FixtureBuilder fb = new FixtureBuilder();
			fixtures = fb.getFixtures();
			fix = 0;


			try {
				foreach (Fixture Fixture in fixtures) {
					FixtureBox usercontrol = new FixtureBox(Fixture.Home_Team, Fixture.Away_Team, Fixture.Date, getTeamBadge(fixtures[fix].Home_Team), getTeamBadge(fixtures[fix].Away_Team));
					MainFlowLayoutPanel.Controls.Add(usercontrol);
					usercontrol.Dock = DockStyle.Fill;
					usercontrol.Visible = false;
					fixtureBoxList.Add(usercontrol);
					fix = fix + 1;

				}

				fixtureBoxList[0].Visible = true;
				fixtureBoxList[0].backButton.Enabled = false;
				this.submitButton.Visible = false;
				this.editButton.Visible = false;

			} catch (NullReferenceException ex) {
				Interaction.MsgBox("No fixtures available");
			}

			if (fix != 10) {
				this.MainFlowLayoutPanel.Controls.Clear();
				fixtureBoxList.Clear();
				Interaction.MsgBox("There are currently no fixtures to display");
			}

		}


		private void submitButton_Click(System.Object sender, System.EventArgs e)
		{
			My.MyProject.Forms.ProgressBarForm.Show();

			this.SubmitPredictions();

			My.MyProject.Forms.ProgressBarForm.Close();

			Interaction.MsgBox("Predictions submitted");
			submitGroupBox.Visible = false;
			this.MainFlowLayoutPanel.Controls.Clear();


		}

		private void editButton_Click_1(System.Object sender, System.EventArgs e)
		{
			List<Control> list = new List<Control>();

			foreach (object control_loopVariable in MainFlowLayoutPanel.Controls) {
				control = control_loopVariable;
				if (control is FixtureReviewUserControl) {
					list.Add(control);
				}
			}

			foreach (Control control_loopVariable in list) {
				control = control_loopVariable;
				MainFlowLayoutPanel.Controls.Remove(control);

			}

			foreach (Fixture fixture in fixtures) {
				FixtureReviewUserControl review = new FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team);
				this.MainFlowLayoutPanel.Controls.Remove(review);

			}

			fixtureBoxList[0].Visible = true;
			fixtureBoxList[0].backButton.Enabled = false;
			fixtureNo = 0;

			this.submitGroupBox.Visible = false;
			this.submitButton.Visible = false;
			this.editButton.Visible = false;
		}

		private void betSlipButton_Click_1(System.Object sender, System.EventArgs e)
		{
			this.MainFlowLayoutPanel.Controls.Clear();
			FixtureBuilder fb = new FixtureBuilder();
			fixtures = fb.getBet();

			this.submitGroupBox.Visible = false;

			try {
				foreach (Fixture fixture in fixtures) {
					FixtureReviewUserControl review = new FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team);
					this.MainFlowLayoutPanel.Controls.Add(review);

					if (fixture.Prediction == 1) {
						review.homeFixtureLabel.BackColor = Color.LightGreen;
					}
					if (fixture.Prediction == 2) {
						review.awayFixtureLabel.BackColor = Color.LightGreen;
					}
					if (fixture.Prediction == 3) {
						review.drawLabel.BackColor = Color.LightGreen;
					}
				}

			} catch (NullReferenceException ex) {
				Interaction.MsgBox("Not enough users have entered their predictions yet. Please try later");
			}
		}

		private void whoBetButton_Click(System.Object sender, System.EventArgs e)
		{
			this.MainFlowLayoutPanel.Controls.Clear();
			FixtureBuilder fb = new FixtureBuilder();
			userList = fb.getUsers();
			submittedList = fb.usersSubmitted();

			this.MainFlowLayoutPanel.Enabled = true;
			this.MainFlowLayoutPanel.Visible = true;
			this.submitGroupBox.Visible = false;

			foreach (string user in userList) {
				UserPredictionsUserControl review = new UserPredictionsUserControl(user);
				this.MainFlowLayoutPanel.Controls.Add(review);

				if (submittedList.Contains(user)) {
					review.submittedLabel.BackColor = Color.LightGreen;
					review.submittedLabel.Text = "Submitted";
				}

			}
		}

		private void predictionLeagueButton_Click(System.Object sender, System.EventArgs e)
		{
			this.MainFlowLayoutPanel.Controls.Clear();
			FixtureBuilder fb = new FixtureBuilder();
			userResultList = fb.getScores();
			Dictionary<int, string> userDictionary = new Dictionary<int, string>();
			userDictionary = fb.getIdsandUsers();

			this.MainFlowLayoutPanel.Enabled = true;
			this.MainFlowLayoutPanel.Visible = true;
			this.submitGroupBox.Visible = false;

			string name = "";

			foreach (Users user in userResultList) {
				userDictionary.TryGetValue(user.UserID, out name);
				PredictionLeagueUserControl review = new PredictionLeagueUserControl(name, user.correct);
				this.MainFlowLayoutPanel.Controls.Add(review);

			}
		}

		private void logOutButton_Click(System.Object sender, System.EventArgs e)
		{
			this.MainFlowLayoutPanel.Controls.Clear();
			this.enterPredictionsButton.Text = "";
			this.betSlipButton.Text = "";
			this.whoBetButton.Text = "";
			this.predictionLeagueButton.Text = "";
			this.myAccountButton.Text = "";
			this.logOutButton.Text = "";
			this.enterPredictionsButton.Enabled = false;
			this.betSlipButton.Enabled = false;
			this.whoBetButton.Enabled = false;
			this.predictionLeagueButton.Enabled = false;
			this.myAccountButton.Enabled = false;
			this.logOutButton.Enabled = false;
			this.coverTitleLabel.Visible = false;
			LoginUserControl fLoginUserControl = new LoginUserControl();
			this.MainFlowLayoutPanel.Controls.Add(fLoginUserControl);
			this.AcceptButton = fLoginUserControl.loginButton;
		}

		public void makeFixtureVisible(bool forward)
		{
			try {

				if ((forward)) {
					int previousFixtureNo = fixtureNo;
					fixtureNo = fixtureNo + 1;
					fixtureBoxList[previousFixtureNo].Visible = false;

					if (fixtureNo == 9) {
						fixtureBoxList[fixtureNo].nextButton.Text = "Review >";
						fixtureBoxList[fixtureNo].nextButton.BackColor = Color.LightGreen;
					}

					if (fixtureBoxList[previousFixtureNo].homeRadioButton.Checked == true) {
						fixtures[previousFixtureNo].Prediction = 1;
					}
					if (fixtureBoxList[previousFixtureNo].awayRadioButton.Checked == true) {
						fixtures[previousFixtureNo].Prediction = 2;
					}
					if (fixtureBoxList[previousFixtureNo].drawRadioButton.Checked == true) {
						fixtures[previousFixtureNo].Prediction = 3;
					}

				} else {
					fixtureNo = fixtureNo - 1;
					fixtureBoxList[fixtureNo + 1].Visible = false;

				}

				fixtureBoxList[fixtureNo].Visible = true;

			} catch (ArgumentOutOfRangeException ex) {
				Interaction.MsgBox("The full set of fixtures is not available. Please email complaints to michael.cole@advancedcomputersoftware.com");
			}
		}

		public int getFixtureNo()
		{
			return fixtureNo;
		}


		public void reviewPredictions()
		{
			int previousFixtureNo = fixtureNo;
			fixtureNo = fixtureNo + 1;
			fixtureBoxList[fixtureNo - 1].Visible = false;

			if (fixtureBoxList[previousFixtureNo].homeRadioButton.Checked == true) {
				fixtures[previousFixtureNo].Prediction = 1;
			}
			if (fixtureBoxList[previousFixtureNo].awayRadioButton.Checked == true) {
				fixtures[previousFixtureNo].Prediction = 2;
			}
			if (fixtureBoxList[previousFixtureNo].drawRadioButton.Checked == true) {
				fixtures[previousFixtureNo].Prediction = 3;
			}

			this.MainFlowLayoutPanel.Enabled = true;
			this.MainFlowLayoutPanel.Visible = true;
			this.submitGroupBox.Visible = true;
			this.submitButton.Visible = true;
			this.editButton.Visible = true;


			foreach (Fixture fixture in fixtures) {
				FixtureReviewUserControl review = new FixtureReviewUserControl(fixture.Home_Team, fixture.Away_Team);
				this.MainFlowLayoutPanel.Controls.Add(review);

				if (fixture.Prediction == 1) {
					review.homeFixtureLabel.BackColor = Color.LightGreen;
				}
				if (fixture.Prediction == 2) {
					review.awayFixtureLabel.BackColor = Color.LightGreen;
				}
				if (fixture.Prediction == 3) {
					review.drawLabel.BackColor = Color.LightGreen;
				}
			}

		}

		public Image getTeamBadge(string team)
		{
			Image badge = PlaceYourBets.My.Resources.Resources.icon2;
			switch (team) {
				case "Arsenal":
					badge = PlaceYourBets.My.Resources.Resources.ARS;
					break;
				case "Aston Villa":
					badge = PlaceYourBets.My.Resources.Resources.AST;
					break;
				case "Burnley":
					badge = PlaceYourBets.My.Resources.Resources.BUR;
					break;
				case "Chelsea":
					badge = PlaceYourBets.My.Resources.Resources.CHE;
					break;
				case "Crystal Palace":
					badge = PlaceYourBets.My.Resources.Resources.CRY;
					break;
				case "Everton":
					badge = PlaceYourBets.My.Resources.Resources.EVE;
					break;
				case "Hull City":
					badge = PlaceYourBets.My.Resources.Resources.HUL;
					break;
				case "Leicester City":
					badge = PlaceYourBets.My.Resources.Resources.LEI;
					break;
				case "Liverpool":
					badge = PlaceYourBets.My.Resources.Resources.LIV;
					break;
				case "Manchester City":
					badge = PlaceYourBets.My.Resources.Resources.MNC;
					break;
				case "Manchester United":
					badge = PlaceYourBets.My.Resources.Resources.MNU;
					break;
				case "Newcastle United":
					badge = PlaceYourBets.My.Resources.Resources.NWC;
					break;
				case "Queens Park Rangers":
					badge = PlaceYourBets.My.Resources.Resources.QPR;
					break;
				case "Southampton":
					badge = PlaceYourBets.My.Resources.Resources.SOU;
					break;
				case "Stoke City":
					badge = PlaceYourBets.My.Resources.Resources.STO;
					break;
				case "Sunderland":
					badge = PlaceYourBets.My.Resources.Resources.SUN;
					break;
				case "Swansea City":
					badge = PlaceYourBets.My.Resources.Resources.SWA;
					break;
				case "Tottenham Hotspur":
					badge = PlaceYourBets.My.Resources.Resources.TOT;
					break;
				case "West Bromwich Albion":
					badge = PlaceYourBets.My.Resources.Resources.WBA;
					break;
				case "West Ham United":
					badge = PlaceYourBets.My.Resources.Resources.WHU;
					break;
				default:
					badge = PlaceYourBets.My.Resources.Resources.icon2;

					break;
			}

			return badge;

		}


		public void SubmitPredictions()
		{
			bool submitted = false;


			while ((!submitted)) {

				foreach (Fixture fixture in fixtures) {
					My.MyProject.Forms.ProgressBarForm.ProgressBar1.PerformStep();
					string UrlBase = "http://www.predictresults.co.uk/API/InsertDataUsingSP.php?userid=";
					string user = currentUser.ToString();
					string fixtureID = fixture.ID.ToString();
					string prediction = fixture.Prediction.ToString();
					string url = UrlBase + user + "&fixtureid=" + fixtureID + "&prediction=" + prediction;
					HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
					HttpWebResponse response = (HttpWebResponse)request.GetResponse();

					//Get the stream associated with the response. 
					Stream receiveStream = response.GetResponseStream();
					StreamReader readStream = new StreamReader(receiveStream, Encoding.UTF8);

					response.Close();
					readStream.Close();

				}

				submitted = true;

			}
		}


	}
}
